linux/LibSensors.c

```diff
@@ -105,99 +99,142 @@ void LibSensors_cleanup(void) {
 #endif /* BUILD_STATIC */
 }
 
+static int tempDriverPriority(const sensors_chip_name* chip) {
+   static const struct TempDriverDefs {
+      const char* prefix;
+      int priority;
+   } tempDrivers[] =  {
+      { "coretemp",    1, },
+      { "via_cputemp", 1, },
+      { "cpu_thermal", 1, },
+      { "k10temp",     1, },
+      { "zenpower",    1, },
+      /* Low priority drivers */
+      { "acpitz",      0, },
+   };
+
+   for (size_t i = 0; i < ARRAYSIZE(tempDrivers); i++)
+      if (String_eq(chip->prefix, tempDrivers[i].prefix))
+         return tempDrivers[i].priority;
+
+   return -1;
+}
+
 void LibSensors_getCPUTemperatures(CPUData* cpus, unsigned int cpuCount) {
-   for (unsigned int i = 0; i <= cpuCount; i++)
-      cpus[i].temperature = NAN;
+   double data[cpuCount + 1];
+#ifdef __clang_analyzer__
+   memset(data, -1, ARRAYSIZE(data)); /* please clang analyzer */
+#endif
+   for (size_t i = 0; i < ARRAYSIZE(data); i++)
+      data[i] = NAN;
```

Comment:
Why not splitting this `NAN` assignment into a separate loop? Should make the Clang Analyzer just as happy …

Reply:
What do you mean by separate loop?
The analyzer is just dump and thinks `0 < ARRAYSIZE(data)` might be `false`.

Reply:
Due to the VLA `data` this actually might be zero, when overflowing with `cpuCount = MAX_INT` due to the implicit conversion from `unsigned int` to `signed int` for the VLA length argument.

Reply:
added an `assert()` and used `cpuCount + 1` as upper bound to please clang analyzer